src/cursor/core_cursor.ts

```diff
@@ -799,7 +762,7 @@ function nextFunction(self: any, callback: Callback) {
     return setCursorDeadAndNotified(self, callback);
   } else if (
     self.cursorState.cursorIndex === self.cursorState.documents.length &&
-    !Long.ZERO.equals(self.cursorState.cursorId)
+    !Long.ZERO.equals(self.cursorState.cursorId!)
```

Comment:
there should always be a cursorId at this point, otherwise we should bail much earlier

Reply:
Yea I fixed with a check, but if only Long.equals took undefined and just returned false wouldn't need `!` 🤷

Reply:
true, but I think in terms of defensive coding this forces us to check for a `cursorId` much earlier which is probably a good thing